packages/openapi-v3-types/src/openapi-v3-spec-types.ts

```diff
@@ -0,0 +1,395 @@
+  schema?: SchemaObject | ReferenceObject;
+  examples?: [ExampleObject | ReferenceObject];
+  example?: ExampleObject | ReferenceObject;
+  encoding?: EncodingObject;
+}
+
+/**
+ * Describes an encoding object, copied from 'openapi3-ts' due to overriding
+ * its `ParameterObject`
+ */
+export interface EncodingObject extends ISpecificationExtension {
+  // [property: string]: EncodingPropertyObject;
+  [property: string]: EncodingPropertyObject | ExtensionValue; // Hack for allowing ISpecificationExtension
+}
+
+/**
+ * Describes an encoding object, copied from 'openapi3-ts' due to overriding
+ * its `ParameterObject`
+ */
+export interface EncodingPropertyObject {
+  contentType?: string;
+  headers?: {[key: string]: HeaderObject | ReferenceObject};
+  style?: string;
+  explode?: boolean;
+  allowReserved?: boolean;
+  [key: string]: ExtensionValue; // (any) = Hack for allowing ISpecificationExtension
+}
+
+/**
+ * Describes an encoding object, copied from 'openapi3-ts' due to overriding
+ * its `ParameterObject`
+ */
+export interface HeaderObject extends ParameterObject {}
+
+/**
+ * Describes a single operation parameter.
+ * A unique parameter is defined by a combination of a name and location.
+ * <p>Specification
+ * https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#parameter-object
+ */
+export interface ParameterObject extends ISpecificationExtension {
+  name: string;
+  in: ParameterLocation;
+  description?: string;
+  required?: boolean;
+  deprecated?: boolean;
+  allowEmptyValue?: boolean;
+
+  style?: ParameterStyle;
+  explode?: boolean;
+  allowReserved?: boolean;
+  schema?: SchemaObject | ReferenceObject;
+  examples?: {[param: string]: ExampleObject | ReferenceObject};
+  example?: ExtensionValue;
+  content?: ContentObject;
+}
+
+/**
+ * Describes an encoding object, copied from 'openapi3-ts' due to overriding
+ * its `ParameterObject`
+ */
+export interface ExampleObject {
+  summary?: string;
+  description?: string;
+  value?: ExtensionValue;
+  externalValue?: string;
+  [property: string]: ExtensionValue; // Hack for allowing ISpecificationExtension
+}
+
+export interface ReferenceObject {
+  $ref: string;
+}
+
+/**
+ * Describes an encoding object, copied from 'openapi3-ts' due to overriding
+ * its `SchemaObject`
+ */
+export interface DiscriminatorObject {
+  propertyName: string;
+  mapping?: {[key: string]: string};
+}
+
+/**
+ * Describes an encoding object, copied from 'openapi3-ts' due to overriding
+ * its `SchemaObject`
+ */
+export interface XmlObject extends ISpecificationExtension {
+  name?: string;
+  namespace?: string;
+  prefix?: string;
+  attribute?: boolean;
+  wrapped?: boolean;
+}
+
+/**
+ * Describes an encoding object, copied from 'openapi3-ts' due to overriding
+ * its `SchemaObject`
+ */
+export interface ExternalDocumentationObject extends ISpecificationExtension {
+  description?: string;
+  url: string;
+}
+
+//  Specification Extensions
+//   ^x-
```

Comment:
This is not a properly documented interface; please add a TSDoc block here instead.